packages/library-base/src/nativeDarwin/kotlin/io/realm/kotlin/internal/platform/SystemUtils.kt

```diff
@@ -137,4 +148,50 @@ private fun preparePath(directoryPath: String, dir: NSURL) {
     }
 }
 
+public fun assetFilePath(assetFileName: String): String? {
+    val pathParts: List<String> = assetFileName.split("[.|/]".toRegex())
+    val path: String? = NSBundle.mainBundle.pathForResource(pathParts[0], pathParts[1])
```

Comment:
Wouldn't it fail with "realm/db.realm"? The split would generate more than 2 parts, and the resource loading would fail to fetch the file 🤔

Reply:
Yep. Good catch. I have added some tests that uses subdirs.